python/lsst/ts/eas/eas_csc.py

```
@@ -133,27 +137,61 @@ async def handle_summary_state(self) -> None:
-        else:
-            self.enabled_event.clear()
-
         if self.disabled_or_enabled:
-            if self.m1m3_thermal_task.done():
+            # Start the thermal task if not already started
+            if self.m1m3_thermal_task.done() and self.simulation_mode == 0:
                 self.m1m3_thermal_task = asyncio.create_task(self.run_control())
 
+            # Connect if not already connected
             if not self.connected:
                 await self.connect()
         else:
+
+            # Cancel the thermal task
             if not self.m1m3_thermal_task.done():
                 self.m1m3_thermal_task.cancel()
                 try:
                     await self.m1m3_thermal_task
                 except asyncio.CancelledError:
                     pass
 
+            # Disconnect
             await self.disconnect()
 
+        # Set the state of the enabled event
+        if self.summary_state == salobj.State.ENABLED:
+            self.enabled_event.set()
+        else:
+            self.enabled_event.clear()
+
+    async def end_disable(self, id_data: salobj.BaseDdsDataType) -> None:
+        """End do_disable; called after state changes but before command
+        acknowledged.
+
+        This method acknowledges the do_disable command and stops the
+        M1M3TS control loop.
+
+        Parameters
+        ----------
+        id_data: `CommandIdData`
+            Command ID and data
+        """
+
+        await self.cmd_disable.ack_in_progress(id_data, timeout=SAL_TIMEOUT)
+        try:
+            async with self.control_loop_lock:
+                await asyncio.wait_for(
+                    self.m1m3_stop(),
+                    timeout=M1M3TS_STOP_TIMEOUT,
+                )
+        except TimeoutError:
+            await self.fault(
+                code=THERMAL_SHUTDOWN_ERROR,
+                report="Failed to stop fans/valve.",
+                traceback=traceback.format_exc(),
+            )
```

you might want to raise an exception here to make sure the command fails.